Description: Set the family name to "OCRB" for both Regular and Slanted styles,
 and to "OCRB" plus single-letter suffix for all the others. Set the fontname to
 the family name with the style strings tacked on to the end. Replace "Slanted"
 with "Oblique" as the latter is a style recognized by fontconfig.
Author: Fabian Greffrath <fabian@debian.org>
Last-Update: 2021-08-20

--- a/Makefile.am
+++ b/Makefile.am
@@ -308,7 +308,7 @@ OCRBG.pfb OCRBG.afm OCRBG.ttf OCRBG.otf
 OCRBL.pfb OCRBL.afm OCRBL.ttf OCRBL.otf &: \
   msk-ocrb10l-rmo.pfb ocrb-@ae_arg_okay@.pe ocrbnames.pe ocrb.fea
 	$(AM_V_at)$(TSU_V_OTF)'$(FONTANVIL)' -lang=ff -script \
-	  $(mvp)/ocrb-@ae_arg_okay@.pe 'l' 'L' "Slanted" 0 \
+	  $(mvp)/ocrb-@ae_arg_okay@.pe 'l' '' "Oblique" 0 \
 	  $(mvp)/ocrb.fea $(TSU_V_REDIR)
 
 OCRBS.pfb OCRBS.afm OCRBS.ttf OCRBS.otf &: \
--- a/ocrb-no.pe
+++ b/ocrb-no.pe
@@ -21,7 +21,9 @@ else
   Open("msk-ocrb10"+$1+"-rmo.pfb");
 endif
 
-SetFontNames("OCRB"+$2,"OCRB","OCR B "+$3,"Regular","Matthew Skala (2011-12); based on code by Norbert Schwarz (1986, 2011)","2");
+family="OCRB"+$2
+fontname=family+"-"+StrJoin(StrSplit($3," "),"");
+SetFontNames(fontname,family,"OCR B "+$3,"Regular","Matthew Skala (2011-12); based on code by Norbert Schwarz (1986, 2011)","2");
 SetTTFName(0x409,0,"Matthew Skala (2011, 2021); based on code by Norbert Schwarz (1986, 2011)");
 SetTTFName(0x409,1,"OCR B "+$2);
 SetTTFName(0x409,2,"Regular");
@@ -29,6 +31,7 @@ SetTTFName(0x409,4,"OCR B "+$3);
 SetTTFName(0x409,11,"https://tsukurimashou.osdn.jp/");
 SetTTFName(0x409,16,"OCR B");
 SetTTFName(0x409,17,$3);
+if (Strlen($2)==0 && Strlen($1)>0) $2=ToUpper($1); endif
 
 SelectAll();SetWidth(723);
 "ocrbnames.pe"();
--- a/ocrb-yes.pe
+++ b/ocrb-yes.pe
@@ -21,7 +21,9 @@ else
   Open("msk-ocrb10"+$1+"-rmo.pfb");
 endif
 
-SetFontNames("OCRB"+$2,"OCRB","OCR B "+$3,"Regular","Matthew Skala (2011); based on code by Norbert Schwarz (1986, 2011)","2");
+family="OCRB"+$2
+fontname=family+"-"+StrJoin(StrSplit($3," "),"");
+SetFontNames(fontname,family,"OCR B "+$3,"Regular","Matthew Skala (2011); based on code by Norbert Schwarz (1986, 2011)","2");
 SetTTFName(0x409,0,"Matthew Skala (2011, 2021); based on code by Norbert Schwarz (1986, 2011)");
 SetTTFName(0x409,1,"OCR B "+$2);
 SetTTFName(0x409,2,"Regular");
@@ -29,6 +31,7 @@ SetTTFName(0x409,4,"OCR B "+$3);
 SetTTFName(0x409,11,"https://tsukurimashou.osdn.jp/");
 SetTTFName(0x409,16,"OCR B");
 SetTTFName(0x409,17,$3);
+if (Strlen($2)==0 && Strlen($1)>0) $2=ToUpper($1); endif
 
 SelectAll();SetWidth(723);
 "ocrbnames.pe"();
